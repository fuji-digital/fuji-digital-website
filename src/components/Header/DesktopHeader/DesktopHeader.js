import { StyledDesktopHeader } from "./DesktopHeader.styles";
import { StandardLogo } from "../../Logo/StandardLogo/StandardLogo";

export const DesktopHeader = () => {
  return (
    <StyledDesktopHeader>
      <StandardLogo />
    </StyledDesktopHeader>
  );
};
