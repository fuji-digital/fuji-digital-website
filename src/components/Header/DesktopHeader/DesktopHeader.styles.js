import styled from "styled-components";
import { ContactButton } from "../../ContactButton/ContactButton";

export const StyledDesktopHeader = styled.div`
  padding: 1.5rem 2rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 2rem;
  font-family: "Poppins", sans-serif;
  box-shadow: 0px 1px 2px #ccd1d1;
  margin: 0rem 0rem 2rem 0rem;

  a {
    color: white;
    text-decoration: none;
    font-weight: bold;
    font-size: 1rem;

    @media screen and (min-width: 1200px) {
      font-size: 1.3rem;
    }
  }

  @media screen and (min-width: 768px) {
    padding: 1rem 3rem;
  }
`;

export const StyledMenu = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 2rem;

  @media screen and (min-width: 1200px) {
    gap: 5rem;
  }
`;

export const StyledSayHello = styled(ContactButton)`
  @media screen and (min-width: 768px) {
    grid-column: 1;
    grid-row: 3;
  }
`;
