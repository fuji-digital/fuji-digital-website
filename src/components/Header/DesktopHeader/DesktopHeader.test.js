import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { DesktopHeader } from "./DesktopHeader";

jest.mock(`framer-motion`, () => ({
  ...jest.requireActual("framer-motion"),
  motion: {
    div: jest.fn().mockImplementation(({ children }) => <div>{children}</div>),
  },
}));

describe("Desktop Header", () => {
  it("matches snapshot", () => {
    const component = renderer
      .create(<DesktopHeader handleModalClick={() => {}} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("click on say hello button handle modal click function is called", () => {
    const mockedHandleModalClick = jest.fn();
    render(<DesktopHeader handleModalClick={mockedHandleModalClick} />);

    const contactButton = screen.getByTestId(`contact-button`);

    fireEvent.click(contactButton);

    expect(mockedHandleModalClick).toHaveBeenCalled();
  });
});
