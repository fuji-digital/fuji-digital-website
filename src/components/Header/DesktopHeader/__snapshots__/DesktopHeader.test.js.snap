// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop Header matches snapshot 1`] = `
<div
  className="sc-eqUzNf kDOjW"
>
  <div
    className="sc-fqkwJk ifBEVm"
  >
    <div
      className="sc-kAycRU bqnGkc"
    >
      <img
        alt="Fuji Digital Logo"
        className="sc-iGgVNO bOGwIO"
        src="fuji-digital-logo.svg"
      />
      <img
        alt="Fuji Digital Text Logo"
        className="sc-gsFSjX cBeDTi"
        data-testid="textLogo"
        src="fuji-digital-text-logo-white.svg"
      />
    </div>
    <a
      href="#what-we-do"
    >
      What We Do
    </a>
    <a
      href="#who-are-we"
    >
      Who Are We
    </a>
  </div>
  <button
    className="sc-aYaIB koDWfZ sc-dcJtft jkTjTA"
    data-testid="contact-button"
    onClick={[Function]}
  >
    <div>
      Say Hello
    </div>
    <img
      alt="A black and white arrow"
      className="sc-gEvDqW dTdWeV"
      src="arrow-icon.svg"
    />
  </button>
</div>
`;
